src/librustc_resolve/diagnostics.rs

```diff
@@ -93,6 +93,41 @@ mod bar {
 ```
 "##,
 
+E0254: r##"
+This error indicates that an imported symbol resolved to an already imported
+symbol.
+
+An example of this error:
+
+```
+// crate foo
+pub use foo::bar;
+pub mod foo; // note that there's a `pub` here
+```
+
+```
+// other program
+extern crate foo;
+
+use foo::*; // error, `foo::*` tries to bind `foo::foo` to `foo`
```

Comment:
(we could remove the `foo.rs` entirely by referring to a crate that is always available.  For example by using `core` everywhere that `foo` occurs in my example.  But its probably better to explicitly include the `foo` crate itself, even if it is a trivial crate with no exports.)